TST: failing builds on master

- [ ] 3.4 slow https://travis-ci.org/pandas-dev/pandas/jobs/181534281

```
======================================================================
FAIL: test_bar_log_no_subplots (pandas.tests.plotting.test_frame.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/pandas-dev/pandas/pandas/tests/plotting/test_frame.py", line 1130, in test_bar_log_no_subplots
    tm.assert_numpy_array_equal(ax.yaxis.get_ticklocs(), expected)
  File "/home/travis/build/pandas-dev/pandas/pandas/util/testing.py", line 1079, in assert_numpy_array_equal
    _raise(left, right, err_msg)
  File "/home/travis/build/pandas-dev/pandas/pandas/util/testing.py", line 1062, in _raise
    .format(obj), left.shape, right.shape)
  File "/home/travis/build/pandas-dev/pandas/pandas/util/testing.py", line 1018, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different
numpy array shapes are different
[left]:  (5,)
[right]: (4,)
======================================================================
FAIL: test_bar_log_subplots (pandas.tests.plotting.test_frame.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/pandas-dev/pandas/pandas/tests/plotting/test_frame.py", line 1141, in test_bar_log_subplots
    tm.assert_numpy_array_equal(ax[0].yaxis.get_ticklocs(), expected)
  File "/home/travis/build/pandas-dev/pandas/pandas/util/testing.py", line 1079, in assert_numpy_array_equal
    _raise(left, right, err_msg)
  File "/home/travis/build/pandas-dev/pandas/pandas/util/testing.py", line 1062, in _raise
    .format(obj), left.shape, right.shape)
  File "/home/travis/build/pandas-dev/pandas/pandas/util/testing.py", line 1018, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different
numpy array shapes are different
[left]:  (7,)
[right]: (6,)
======================================================================
FAIL: test_bar_log (pandas.tests.plotting.test_series.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/pandas-dev/pandas/pandas/tests/plotting/test_series.py", line 207, in test_bar_log
    tm.assert_numpy_array_equal(ax.yaxis.get_ticklocs(), expected)
  File "/home/travis/build/pandas-dev/pandas/pandas/util/testing.py", line 1079, in assert_numpy_array_equal
    _raise(left, right, err_msg)
  File "/home/travis/build/pandas-dev/pandas/pandas/util/testing.py", line 1062, in _raise
    .format(obj), left.shape, right.shape)
  File "/home/travis/build/pandas-dev/pandas/pandas/util/testing.py", line 1018, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different
numpy array shapes are different
[left]:  (7,)
[right]: (6,)
----------------------------------------------------------------------
Ran 252 tests in 1103.335s
```

- [x] numpy-dev build: https://travis-ci.org/pandas-dev/pandas/jobs/181534284, closed by #14809 

```
======================================================================
ERROR: test_scalar_na_cmp_corners (pandas.tests.series.test_operators.TestSeriesOperators)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/pandas-dev/pandas/pandas/core/ops.py", line 1170, in na_op
    raise_on_error=True, **eval_kwargs)
  File "/home/travis/build/pandas-dev/pandas/pandas/computation/expressions.py", line 210, in evaluate
    **eval_kwargs)
  File "/home/travis/build/pandas-dev/pandas/pandas/computation/expressions.py", line 63, in _evaluate_standard
    return op(a, b)
TypeError: unsupported operand type(s) for &: 'float' and 'float'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/travis/build/pandas-dev/pandas/pandas/core/internals.py", line 1160, in eval
    result = get_result(other)
  File "/home/travis/build/pandas-dev/pandas/pandas/core/internals.py", line 1129, in get_result
    result = func(values, other)
  File "/home/travis/build/pandas-dev/pandas/pandas/core/ops.py", line 1179, in na_op
    yrav = yrav[mask]
IndexError: boolean index did not match indexed array along dimension 0; dimension is 1 but corresponding boolean dimension is 9
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/travis/build/pandas-dev/pandas/pandas/tests/series/test_operators.py", line 1417, in test_scalar_na_cmp_corners
    self.assertRaises(ValueError, lambda: d.__and__(s, axis='columns'))
  File "/home/travis/build/pandas-dev/pandas/pandas/util/testing.py", line 2380, in assertRaises
    _callable(*args, **kwargs)
  File "/home/travis/build/pandas-dev/pandas/pandas/tests/series/test_operators.py", line 1417, in <lambda>
    self.assertRaises(ValueError, lambda: d.__and__(s, axis='columns'))
  File "/home/travis/build/pandas-dev/pandas/pandas/core/ops.py", line 1223, in f
    return self._combine_series(other, na_op, fill_value, axis, level)
  File "/home/travis/build/pandas-dev/pandas/pandas/core/frame.py", line 3539, in _combine_series
    fill_value=fill_value)
  File "/home/travis/build/pandas-dev/pandas/pandas/core/frame.py", line 3573, in _combine_match_columns
    axes=[left.columns, self.index])
  File "/home/travis/build/pandas-dev/pandas/pandas/core/internals.py", line 3163, in eval
    return self.apply('eval', **kwargs)
  File "/home/travis/build/pandas-dev/pandas/pandas/core/internals.py", line 3057, in apply
    applied = getattr(b, f)(**kwargs)
  File "/home/travis/build/pandas-dev/pandas/pandas/core/internals.py", line 1167, in eval
    result = handle_error()
  File "/home/travis/build/pandas-dev/pandas/pandas/core/internals.py", line 1150, in handle_error
    (repr(other), str(detail)))  # noqa
TypeError: Could not operate array([nan], dtype=object) with block values boolean index did not match indexed array along dimension 0; dimension is 1 but corresponding boolean dimension is 9
----------------------------------------------------------------------
```

--- a/pandas/tests/plotting/test_series.py
+++ b/pandas/tests/plotting/test_series.py
@@ -216,16 +216,22 @@ class TestSeriesPlots(TestPlotBase):
 
         if not self.mpl_le_1_2_1:
             expected = np.hstack((1.0e-04, expected, 1.0e+01))
+        if self.mpl_ge_2_0_0:
+            expected = np.hstack((1.0e-05, expected))
 
         ax = Series([0.1, 0.01, 0.001]).plot(log=True, kind='bar')
-        ymin = 0.00079432823472428218 if self.mpl_ge_2_0_0 else 0.001
+        ymin = 0.0007943282347242822 if self.mpl_ge_2_0_0 else 0.001
         ymax = 0.12589254117941673 if self.mpl_ge_2_0_0 else .10000000000000001
-        self.assertEqual(ax.get_ylim(), (ymin, ymax))
+        res = ax.get_ylim()
+        self.assertAlmostEqual(res[0], ymin)
+        self.assertAlmostEqual(res[1], ymax)
         tm.assert_numpy_array_equal(ax.yaxis.get_ticklocs(), expected)
         tm.close()
 
         ax = Series([0.1, 0.01, 0.001]).plot(log=True, kind='barh')
-        self.assertEqual(ax.get_xlim(), (0.001, ymax))
+        res = ax.get_xlim()
+        self.assertAlmostEqual(res[0], ymin)
+        self.assertAlmostEqual(res[1], ymax)
         tm.assert_numpy_array_equal(ax.xaxis.get_ticklocs(), expected)
 
     @slow

--- a/pandas/tests/plotting/test_series.py
+++ b/pandas/tests/plotting/test_series.py
@@ -218,8 +218,9 @@ class TestSeriesPlots(TestPlotBase):
             expected = np.hstack((1.0e-04, expected, 1.0e+01))
 
         ax = Series([0.1, 0.01, 0.001]).plot(log=True, kind='bar')
+        ymin = 0.00079432823472428218 if self.mpl_ge_2_0_0 else 0.001
         ymax = 0.12589254117941673 if self.mpl_ge_2_0_0 else .10000000000000001
-        self.assertEqual(ax.get_ylim(), (0.001, ymax))
+        self.assertEqual(ax.get_ylim(), (ymin, ymax))
         tm.assert_numpy_array_equal(ax.yaxis.get_ticklocs(), expected)
         tm.close()
 

